src/Terrain/MapMaster.java

```diff
@@ -121,7 +121,9 @@ public Environment getEnvironment(int w, int h)
   /** Returns the Location at the specified location, or null if that Location does not exist. */
   public Location getLocation(XYCoord location)
   {
-    return getLocation(location.xCoord, location.yCoord);
+    if (null != location)
+      return getLocation(location.xCoord, location.yCoord);
+    return null;
```

Comment:
I might just make this a NPE.

Reply:
It doesn't violate the constraints of the function, since it can already return null for a location outside the map.
"Coordinate null" sounds like it's outside the map to me :V